### ISSUE TITLE
Compile Error on Linux: jemalloc again (BB #1141)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#1141]
This issue was migrated from bitbucket.
**Original Reporter:** Xaanor
**Original Date:** 11.10.2015 12:32:17 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1141

<hr>

I have clean Setup with Debian 8 amd64.

```
#!c++

Linux testbox 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt11-1+deb8u4 (2015-09-19) x86_64 GNU/Linux

gcc version 4.9.2 (Debian 4.9.2-10)

```

CMAKE-Output:

```
#!c++

matze@testbox:~/src/oregoncore/build$ cmake ../ -DPREFIX=$HOME -DTOOLS=1
-- Detected 64-bit platform
-- UNIX: Using default configuration directory
-- UNIX: Using default library directory
-- UNIX: Configuring uninstall target
-- UNIX: Created uninstall target
-- GCC: All warnings disabled
-- Found ACE library: /usr/lib/libACE.so
-- Found ACE headers: /usr/include
-- Using mysql-config: /usr/bin/mysql_config
-- Found MySQL library: /usr/lib/x86_64-linux-gnu/libmysqlclient_r.so
-- Found MySQL headers: /usr/include/mysql
-- Found Curses library: /usr/lib/x86_64-linux-gnu/libcurses.so
-- Found Binutils

* OregonCore revision    : 3073+ (622a8fda2890+)
* Build binaries in      : Release mode

* Install core to        : /home/matze/bin
* Install libraries to   : /home/matze/lib
* Install configs to     : /home/matze/etc

* Build world/auth       : Yes (default)
* Build with scripts     : Yes (default)
* Build map extractiors  : Yes

* Build core w/PCH       : Yes (default)
* Build scripts w/PCH    : Yes (default)
* Use SFMT for RNG       : No  (default)
* Show compile-warnings  : No  (default)
* Use coreside debug     : No  (default)
* Build w/ documentation : No  (default)

-- Configuring done
-- Generating done
-- Build files have been written to: /home/matze/src/oregoncore/build
matze@testbox:~/src/oregoncore/build$

```

MAKE-Output:

```
#!c++

matze@testbox:~/src/oregoncore/build$ make
Scanning dependencies of target jemalloc_internals
[  0%] Generating jemallocs_internals
[  0%] Built target jemalloc_internals
Scanning dependencies of target jemalloc
[  1%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/arena.c.o
[  1%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/atomic.c.o
[  1%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/base.c.o
[  1%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/bitmap.c.o
[  1%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/chunk.c.o
[  1%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/chunk_dss.c.o
[  1%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/chunk_mmap.c.o
[  2%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/ckh.c.o
[  2%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/ctl.c.o
[  2%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/extent.c.o
[  2%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/hash.c.o
[  2%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/huge.c.o
[  2%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/jemalloc.c.o
[  2%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/mb.c.o
[  3%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/mutex.c.o
[  3%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/pages.c.o
[  3%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/prof.c.o
[  3%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/quarantine.c.o
[  3%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/rtree.c.o
[  3%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/stats.c.o
[  3%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/tcache.c.o
[  4%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/tsd.c.o
[  4%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/util.c.o
[  4%] Building C object dep/jemalloc/CMakeFiles/jemalloc.dir/src/valgrind.c.o
/home/matze/src/oregoncore/dep/jemalloc/src/valgrind.c:3:4: error: #error "This source file is for Valgrind integration."
 #  error "This source file is for Valgrind integration."
    ^
/home/matze/src/oregoncore/dep/jemalloc/src/valgrind.c:6:31: fatal error: valgrind/memcheck.h: Datei oder Verzeichnis nicht gefunden
 #include <valgrind/memcheck.h>
                               ^
compilation terminated.
dep/jemalloc/CMakeFiles/jemalloc.dir/build.make:583: recipe for target 'dep/jemalloc/CMakeFiles/jemalloc.dir/src/valgrind.c.o' failed
make[2]: *** [dep/jemalloc/CMakeFiles/jemalloc.dir/src/valgrind.c.o] Error 1
CMakeFiles/Makefile2:134: recipe for target 'dep/jemalloc/CMakeFiles/jemalloc.dir/all' failed
make[1]: *** [dep/jemalloc/CMakeFiles/jemalloc.dir/all] Error 2
Makefile:117: recipe for target 'all' failed
make: *** [all] Error 2
matze@testbox:~/src/oregoncore/build$

```

So as you can see, we have a compile error again.

valgrind/memcheck.h is not present anywhere.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 11.10.2015 12:34:31 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1141/#comment-22409065

<hr>

Interesting. Quick fix will be to install valgrind. I will fix this asap, tho.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 11.10.2015 12:43:12 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1141/#comment-22409132

<hr>

Don't include valgrind.c by default in jemalloc - fix #1141

→ <<cset 7e902f6d96e9>>
